test/functional/helpers.lua

```diff
@@ -270,6 +270,13 @@ function module.command(cmd)
   module.request('nvim_command', cmd)
 end
 
+
+-- use for commands which expect nvim to quit
+function module.expect_exit(...)
+  eq("EOF was received from Nvim. Likely the Nvim process crashed.",
+     module.pcall_err(...))
```

Comment:
I guess clients could set up a VimLeave handler, so they know that Nvim is exiting and not crashing. Then we wouldn't need expect_exit() ?

Reply:
> Then we wouldn't need expect_exit() ?

`expect_exit()` was added to assert the actual behavior of nvim: that the channel was closed (eof) before receiving a valid response to the request from the client. More importantly, this means all other requests in the test suite now expect that we get a non-error response, even when we don't check the return value. Implicitly discarding this error was a bug in lua-client that I fixed.